src/stream/src/executor/top_n/top_n_appendonly.rs

```diff
@@ -188,9 +188,15 @@ where
             .await
     }
 
-    async fn handle_watermark(&mut self, _: Watermark) -> Option<Watermark> {
-        // TODO(yuhao): handle watermark
-        None
+    async fn handle_watermark(
+        &mut self,
+        watermark: Watermark,
+    ) -> StreamExecutorResult<Option<Watermark>> {
+        Ok(if watermark.col_idx == self.storage_key_indices[0] {
+            Some(watermark)
```

Comment:
ditto. descending case. I think we need to compare the last key with the watermark here.